### ISSUE TITLE
compile warning: enumeral and non-enumeral type in conditional expression

### ISSUE THREAD

**Issue by alanw0** [kokkos/kokkos#3397]
We are seeing this compile warning with gcc 7.2.0, with c++14 flag enabled.

/path/include/impl/Kokkos_ViewMapping.hpp:210:68: warning: enumeral and non-enumeral type in conditional expression [-Wextra]
                                                          : (r == 7 ? N7
                                                            ~~~~~~~~^~~~
                                                                    : 0)))))));
                                                                    ~~~~

**Comment by dalg24** [comment]
Duplicate of #3341

**Comment by crtrott** [comment]
What are your warning flags?

**Comment by crtrott** [comment]
Ah i see its wextra, we do not promise that to be warning free. Like this particular issue i tried to fix but its an absolute mess partly because nums are different things on different platforms.

**Comment by ndellingwood** [comment]
Addressed by #3422 (which superceded the related PR #3420), marking InDevelop
